stubs/WTForms/wtforms/fields/core.pyi

```diff
@@ -0,0 +1,133 @@
+from collections.abc import Callable, Iterable, Sequence
+from typing import Any, Generic, Protocol, TypeVar, overload
+from typing_extensions import Self, TypeAlias
+
+from markupsafe import Markup
+from wtforms.form import BaseForm
+from wtforms.meta import DefaultMeta, _MultiDictLikeWithGetlist, _SupportsGettextAndNgettext
+
+_FormT = TypeVar("_FormT", bound=BaseForm)
+_FieldT = TypeVar("_FieldT", bound=Field)
+_FormT_contra = TypeVar("_FormT_contra", bound=BaseForm, contravariant=True)
+_FieldT_contra = TypeVar("_FieldT_contra", bound=Field, contravariant=True)
+# It would be nice to annotate this as invariant, i.e. input type and output type
+# needs to be the same, but it will probably be too annoying to use, for now we
+# trust, that people won't use it to change the type of data in a field...
+_Filter: TypeAlias = Callable[[Any], Any]
+
+class _Validator(Protocol[_FormT_contra, _FieldT_contra]):
+    def __call__(self, __form: _FormT_contra, __field: _FieldT_contra) -> object: ...
+
+class _Widget(Protocol[_FieldT_contra]):
+    def __call__(self, field: _FieldT_contra, **kwargs: Any) -> Markup: ...
+
+class Field:
+    errors: Sequence[str]
+    process_errors: Sequence[str]
+    raw_data: list[Any] | None
+    object_data: Any
+    data: Any
+    validators: Sequence[_Validator[Any, Self]]
+    # even though this could be None on the base class, this should
+    # never actually be None in a real field
+    widget: _Widget[Self]
+    do_not_call_in_templates: bool
+    meta: DefaultMeta
+    default: Any | None
+    description: str
+    render_kw: dict[str, Any]
+    filters: Sequence[_Filter]
+    flags: Flags
+    name: str
+    short_name: str
+    id: str
+    label: Label
+    # technically this can return UnboundField, but that is not allowed
```

Comment:
Would it be better to just omit this method? I'm not sure adding it gives us anything.

Reply:
Considering the behavior of `__new__` might change in the future I wanted to leave a breadcrumb about the custom `__new__` so it can maybe be improved in the future. See https://github.com/python/mypy/pull/16020